### PR TITLE
✨ Implement .NET UserTaskConsumer API 

### PR DESCRIPTION
**Pull request by ElRaptorus**
**Changes:**

1. Implement UserTask API.
2. Add missing `FinishManualTask` UseCase.
3. Enhance .NET tests for ManualTasks and UserTasks.

PR: #45

## How can others test the changes?

The .NET client should now be able to handle UserTasks and to finish ManualTasks.

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.).
- [x] I've rebased the `develop` branch with my branch before finishing this PR.
- [x] I've **summarized all changes** in a list above.
- [x] I've mentioned all **PRs, which relate to this one**.
- [x] I've prefixed my Pull Request title is according to [gitmoji guide](https://gitmoji.carloscuesta.me/).

### CODE REVIEW

File: dotnet/src/ConsumerApiClientService.cs [hunk 1]
```diff
@@ -228,74 +228,129 @@ public async Task TriggerSignalEvent(IIdentity identity, string signalName, obje
...
-            {
-                NamingStrategy = new CamelCaseNamingStrategy()
-            };
-            var serializerSettings = new JsonSerializerSettings
+            var url = RestSettings.Paths.ProcessModelUserTasks
+                .Replace(RestSettings.Params.ProcessModelId, processModelId);
+
+            url = $"{RestSettings.Endpoints.ConsumerAPI}/{url}";
+
+            var request = this.CreateRequestMessage(identity, HttpMethod.Get, url);
```
Comment (christianbernds) [inline dotnet/src/ConsumerApiClientService.cs]: Würde alles bis auf die Url in eine `GetUserTasksForUrl` auslagern und in `GetUserTasksForProcessModel` sowie `GetUserTasksForProcessInstance` verwenden. Dann ist die Logik des "GetUserTasks" an 1 Stelle gekapselt.

Kann auch für `GetUserTasksForCorrelation` etc. verwendet werden.

File: dotnet/tests/ConsumerAPIService/ManualTasks/FinishManualTaskTests.cs [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+        public FinishManualTaskTests(ConsumerAPIFixture fixture)
+        {
+            this.fixture = fixture;
+        }
+
+        [Fact]
+        public void FinishManualTask_EmptyParameters_ShouldThrowException()
+        {
+            var ex = Assert.ThrowsAsync<ArgumentNullException>(async () => await this
```
Comment (christianbernds) [inline dotnet/tests/ConsumerAPIService/ManualTasks/FinishManualTaskTests.cs]: variable ex wird nicht benötigt

File: dotnet/tests/ConsumerAPIService/ManualTasks/FinishManualTaskTests.cs [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+                .fixture
+                .ConsumerAPIClient
+                .FinishManualTask(this.fixture.DefaultIdentity, "", "", "")
+            );
+        }
+
+        [Fact]
+        public void FinishManualTask_ProcessInstanceNotFound_ShouldThrowException()
+        {
+            var ex = Assert.ThrowsAsync<Exception>(async () => await this
```
Comment (christianbernds) [inline dotnet/tests/ConsumerAPIService/ManualTasks/FinishManualTaskTests.cs]: variable ex wird nicht benötigt

File: dotnet/tests/ConsumerAPIService/ManualTasks/GetManualTasksForCorrelationTests.cs [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+        public GetManualTasksForCorrelationTests(ConsumerAPIFixture fixture)
+        {
+            this.fixture = fixture;
+        }
+
+        [Fact]
+        public void GetManualTasksForCorrelation_EmptyParameters_ShouldThrowException()
+        {
+            var ex = Assert.ThrowsAsync<ArgumentNullException>(async () => await this
```
Comment (christianbernds) [inline dotnet/tests/ConsumerAPIService/ManualTasks/GetManualTasksForCorrelationTests.cs]: variable ex wird nicht benötigt

File: dotnet/tests/ConsumerAPIService/ManualTasks/GetManualTasksForCorrelationTests.cs [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+                .fixture
+                .ConsumerAPIClient
+                .GetManualTasksForCorrelation(this.fixture.DefaultIdentity, "")
+            );
+        }
+
+        [Fact]
+        public void GetManualTasksForCorrelation_ProcessModelNotFound_ShouldThrowException()
+        {
+            var ex = Assert.ThrowsAsync<Exception>(async () => await this
```
Comment (christianbernds) [inline dotnet/tests/ConsumerAPIService/ManualTasks/GetManualTasksForCorrelationTests.cs]: variable ex wird nicht benötigt

File: dotnet/src/ConsumerApiClientService.cs [hunk 6]
```diff
@@ -228,74 +228,129 @@ public async Task TriggerSignalEvent(IIdentity identity, string signalName, obje
...
-            var hasNoIdentity = identity == null || identity.Token == null;
-            if (hasNoIdentity)
-            {
-                throw new UnauthorizedAccessException();
-            }
+            var url = RestSettings.Paths.ProcessInstanceUserTasks
+                .Replace(RestSettings.Params.ProcessInstanceId, processInstanceId);
 
-            var message = new HttpRequestMessage();
+            url = $"{RestSettings.Endpoints.ConsumerAPI}/{url}";
```
Comment (cmg-dev) [inline dotnet/src/ConsumerApiClientService.cs]:
```suggestion
            var urlWithEndpoint = $"{RestSettings.Endpoints.ConsumerAPI}/{url}";
```

File: dotnet/src/ConsumerApiClientService.cs [hunk 7]
```diff
@@ -228,74 +228,129 @@ public async Task TriggerSignalEvent(IIdentity identity, string signalName, obje
...
 
-            message.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
-            message.Headers.Authorization = new AuthenticationHeaderValue("Bearer", identity.Token);
+            var request = this.CreateRequestMessage(identity, HttpMethod.Get, url);
```
Comment (cmg-dev) [inline dotnet/src/ConsumerApiClientService.cs]:
```suggestion
            var request = this.CreateRequestMessage(identity, HttpMethod.Get, urlWithEndpoint);
```

File: dotnet/src/ConsumerApiClientService.cs [hunk 8]
```diff
@@ -228,74 +228,129 @@ public async Task TriggerSignalEvent(IIdentity identity, string signalName, obje
...
-            throw new NotImplementedException();
-        }
+            var url = RestSettings.Paths.CorrelationUserTasks
+                .Replace(RestSettings.Params.CorrelationId, correlationId);
 
-        public Task<UserTaskList> GetUserTasksForCorrelation(IIdentity identity, string correlationId)
-        {
-            throw new NotImplementedException();
-        }
+            url = $"{RestSettings.Endpoints.ConsumerAPI}/{url}";
```
Comment (cmg-dev) [inline dotnet/src/ConsumerApiClientService.cs]:
```suggestion
            var urlWithEndpoint = $"{RestSettings.Endpoints.ConsumerAPI}/{url}";
```

File: dotnet/src/ConsumerApiClientService.cs [hunk 9]
```diff
@@ -228,74 +228,129 @@ public async Task TriggerSignalEvent(IIdentity identity, string signalName, obje
...
 
-        public Task<UserTaskList> GetUserTasksForProcessModelInCorrelation(IIdentity identity, string processModelId, string correlationId)
-        {
-            throw new NotImplementedException();
+            var request = this.CreateRequestMessage(identity, HttpMethod.Get, url);
```
Comment (cmg-dev) [inline dotnet/src/ConsumerApiClientService.cs]:
```suggestion
            var request = this.CreateRequestMessage(identity, HttpMethod.Get, urlWithEndpoint);
```

File: dotnet/src/ConsumerApiClientService.cs [hunk 10]
```diff
@@ -228,74 +228,129 @@ public async Task TriggerSignalEvent(IIdentity identity, string signalName, obje
...
 
-        public Task<UserTaskList> GetWaitingUserTasksByIdentity(IIdentity identity)
+        public async Task<UserTaskList> GetUserTasksForProcessModelInCorrelation(IIdentity identity, string processModelId, string correlationId)
         {
-            throw new NotImplementedException();
+            var url = RestSettings.Paths.ProcessModelCorrelationUserTasks
+                .Replace(RestSettings.Params.ProcessModelId, processModelId)
+                .Replace(RestSettings.Params.CorrelationId, correlationId);
+
+            url = $"{RestSettings.Endpoints.ConsumerAPI}/{url}";
```
Comment (cmg-dev) [inline dotnet/src/ConsumerApiClientService.cs]:
```suggestion
            var urlWithEndpoint = $"{RestSettings.Endpoints.ConsumerAPI}/{url}";
```

File: dotnet/src/ConsumerApiClientService.cs [hunk 11]
```diff
@@ -228,74 +228,129 @@ public async Task TriggerSignalEvent(IIdentity identity, string signalName, obje
...
+
+            var request = this.CreateRequestMessage(identity, HttpMethod.Get, url);
```
Comment (cmg-dev) [inline dotnet/src/ConsumerApiClientService.cs]:
```suggestion
            var request = this.CreateRequestMessage(identity, HttpMethod.Get, urlWithEndpoint);
```

File: dotnet/src/ConsumerApiClientService.cs [hunk 12]
```diff
@@ -228,74 +228,129 @@ public async Task TriggerSignalEvent(IIdentity identity, string signalName, obje
...
+            return parsedResult;
         }
 
-        public Task FinishUserTask(IIdentity identity, string processInstanceId, string correlationId, string userTaskInstanceId, UserTaskResult userTaskResult)
+        public async Task<UserTaskList> GetWaitingUserTasksByIdentity(IIdentity identity)
         {
-            throw new NotImplementedException();
+            var url = RestSettings.Paths.GetOwnUserTasks;
+
+            url = $"{RestSettings.Endpoints.ConsumerAPI}/{url}";
```
Comment (cmg-dev) [inline dotnet/src/ConsumerApiClientService.cs]: same

**Comment by ElRaptorus** [comment]
@cmg-dev @christianbernds done

**Pull request merged**